tools/shared/src/main/scala/org/scalajs/core/tools/javascript/JSDesugaring.scala

```diff
@@ -336,19 +338,31 @@ object JSDesugaring {
         case Assign(select @ JSDotSelect(qualifier, item), rhs) =>
           unnest(qualifier, rhs) { (newQualifier, newRhs, env0) =>
             implicit val env = env0
-            js.Assign(
-                js.DotSelect(transformExpr(newQualifier), item)(select.pos),
-                transformExpr(newRhs))
+            if (isStrongMode) {
+              js.Apply(js.VarRef(js.Ident("$jsAssign")), List(
+                  transformExpr(newQualifier), js.StringLiteral(item.name),
+                  transformExpr(newRhs)))
```

Comment:
Strong mode is unlikely to go through Closure Advanced mode for the next few years, if ever. So I think we're quite safe about that ... ^^